scripts/apptester/apptester.py

```diff
@@ -245,26 +245,29 @@ def build_applications(self, name=None):
                 else:
                     self.result_matrix[app] = TestStatus.FAILED
 
+        for app in self.result_matrix:
+            app_data = '{} ({}) for {}:'.format(app.get_name(), app.get_language(),
+                        app.get_platform())
+            build_result = [app_data, self.result_matrix[app], 'N/A']
+            self.table_data.append(build_result)
+
     def build_android_java_demo(self):
-        """
-        """
-        self.result_matrix = {}
         output = self.sandboxframe.get_demo_projects()
         for item in output:
             try:
                 if 'java' in item['id'] or 'android' in item['id']:
-
-                    if item['destBinaryFile'] == '':
+                    destBinaryFile = item.get('destBinaryFile', None)
+                    if not destBinaryFile:
                         continue
-
                     else:
                         build_app = self.sandboxframe.build_android_java_demo(item['id'])
+                        build_app_result = self.sandboxframe.is_build_successful(item['id'])
 
-                        if self.sandboxframe.is_build_successful(item['id']):
+                        if build_app_result:
                             self.result_matrix[item['name']] = TestStatus.PASSED
                             print 'Building {}:\n{}'.format(item['name'], build_app)
 
-                        elif not self.sandboxframe.is_build_successful(item['id']):
+                        elif not build_app_result:
```

Comment:
There are two options only: True and False.
So:
if build_app_result:
...
else:
...